### ISSUE TITLE
Random map setup doesn't allow reselection of bigger "computer only players" numbers

### ISSUE THREAD

**Issue by mikeiit** [vcmi/vcmi#4881]
**Describe the bug**
Cannot reselect higher "computer only players" number once I select a lower one

**Game logs**
These logs might be relevant - 
```
[2024-Nov-06 15:00:04.261873] DEBUG [runNetwork] global - Blocking buttons
[2024-Nov-06 15:00:04.261873] DEBUG [runNetwork] global - Blocking buttons
[2024-Nov-06 15:00:04.261873] DEBUG [runNetwork] global - Blocking buttons
```

**To Reproduce**
1. Start a new game
2. Go to random map setup
3. Change "Human or computer players" to 2
4. Observe. "Computer only players" should be 6 by default
5. Change "Computer only players" to 3
6. Then try reselect anything between 4 and 6

**Expected behavior**
IMO it should allow me to as long as the sum of "Human or computer players" and "Computer only players" is still smaller than 8

**Actual behavior**
All buttons greater than 3 after step 5 are disabled. Cannot reselect any. 

**Did it work earlier?**
Not sure.

**Screenshots**
![{71678089-B71D-416E-BA6E-DB71ED024F56}](https://github.com/user-attachments/assets/a9e40020-7e85-4caa-b2c8-d2a6407284fb)

**Version**
 - OS: Windows 11 Pro
 - Version: VCMI 1.6, latest code from Nov 6th

**Additional context**
N/A

**Comment by misiokles** [comment]
Isn't this OH3 behaviour? 2 is sum of human and comp players

**Comment by mikeiit** [comment]
I don't think so. I tried this in complete, or maybe the steam HD version. I was able to select computer only players count up to 6 when the human and CPU players count is 2. My understanding is the sum of those two should not be bigger than 8. That is enough.

**Comment by mikeiit** [comment]
Although there is a workaround. If I click random button again, the count buttons seem to be reset. Then I can select higher count again.

**Comment by mikeiit** [comment]
> Isn't this OH3 behaviour? 2 is sum of human and comp players

I think the sum of those two counts needs to be smaller than the total player limit, either 8 or whatever is set in the current template
